handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+    !!!note
+        After this step SystemLink will migrate your test steps, results, and products from MongoDB to PostgreSQL. Depending on the size of your data set this process may take some time.
+
+1. Verify your new SystemLink Server has all the expected migrated data.
+
+## Upgrading Multi Node Configurations
+
+The following workflow describes upgrading a SystemLink Server instance that has been configured to use a dedicated MongoDB server or replica set, a dedicated PostgreSQL server or replica set, and a dedicated file store such as AWS S3.
+
+1. Backup your SystemLink Server, MongoDB server or replica set, PostgreSQL server or replica set, and file store.
+
+    !!!note
+        If S3 is used as your file store the backup step is not needed since this managed internally by AWS.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --tags --dir D:\migration`.
+
+    !!!note
+        While most of the data is external to SystemLink, tag current values are stored on the app server. Therefore this `nislmigrate` argument flag must be used to ensure that data is not lost during the migration.
+
+1. Detach `D:\`.
+
+    !!!note
+        At this point in the workflow your original SystemLink Server is in an operable state. Be aware if any new data is created or ingested by the SystemLink server at this time it will not be available to your new server.
+
+1. Provision a new Windows server for SystemLink.
+
+1. Provision a MongoDB server or replica set from the backup previously created.
+
+1. Provision a PostGres server or replica set from the backup previously created.
+
+    !!!note
+        The previous steps prescribe using a newly created instance of your MongoDB or PostgreSQL servers. This is because SystemLink may change the internal schema of these databases when services start post upgrade. If you do not start with a instance from a backup you will be unable to revert the MongoDB collections and PostgreSQL tables into the schemas needed for the older version of SystemLink.
+
+1. Install and configure the new version of SystemLink Server.
+
+1. Configure SystemLink to use the newly created MongoDB server or replica set, PostgreSQL server or replica set, and existing file store.
+
+1. Install `nislmigrate` on your new SystemLink Server.
+
+1. Attach the `D:\` volume used to capture data from your original SystemLink server.
+
+1. Run the command `nislmigrate restore --tags --dir D:\migration`.
+
+1. Verify your new SystemLink Server has all the expected migrated data.
+
+## Seamless Cut-over
+
+Since managed test systems are connected SystemLink it is desirable to approach upgrades or migrations such that test systems can connect to the new instance of SystemLink without manual intervention. To accomplish this the following conditions must be met.
```

Comment:
Since managed test systems are connected SystemLink  test systems can connect to the new instance of SystemLink without manual intervention. To accomplish this the following conditions must be met.

To ensure that managed test systems can connect to the new instance of SystemLink without manual intervention (automatically?), check that the following conditions are met.